KubeArmor/BPF/shared.h

```diff
@@ -508,22 +516,64 @@ static inline int match_and_enforce_path_hooks(struct path *f_path, u32 id,
   }
 
 decision:
-
-  if (id == dpath) { // Path Hooks
+  if (id == dpath) { // Path Hooks  
     if (match) {
       if (val && (val->filemask & RULE_OWNER)) {
-        if (!is_owner_path(f_path->dentry)) {
-          retval = -EPERM;
+        struct dentry *dent ;
+        if(eventID  == _FILE_MKNOD || eventID == _FILE_MKDIR){
+          dent = BPF_CORE_READ(f_path , dentry , d_parent);
         } else {
-          return 0;
+          dent = f_path->dentry ;
+        }
+        
+        if (!is_owner_path(dent , &oid)) {
+                if(val->filemask & RULE_DENY) {
+                    retval = DENY;
+                  }
+                else if(val->filemask & RULE_AUDIT){
+                    retval = AUDIT; 
+                } else {
+                      //check for matched owner only allow policies 
+                      bpf_map_update_elem(&bufk, &two, z, BPF_ANY);
+                      pk->path[0] = dfile;
+                      struct data_t *allow = bpf_map_lookup_elem(inner, pk);
+                      if (allow) {
+                          if (allow->processmask == BLOCK_POSTURE) {
+                              retval = DENY;
+                            }
+                      else {
+                          retval = AUDIT;
+                        }
+                      }   
+                    }
+        }        
+         else { 
+              // owner + readonly ( Alerts if read only rule is applied )
+              if (val && (val->filemask & RULE_READ) && !(val->filemask & RULE_WRITE))
+              {
+                if(val->filemask & RULE_DENY) {
+                  retval = DENY;
+                }
+                else if(val->filemask & RULE_AUDIT){
+                  retval = AUDIT;          
+                } 
```

Comment:
We would need to check allow based default posture here too right?
Or is it handled already, since default case isn't to allow?

Reply:
Yep, you are right there could be a case with the `owner + readonly + allow`  policy.